src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1113,13 +1114,11 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
   }
 
   private def withMethod(workspaceContext: WorkspaceContext, methodNamespace: String, methodName: String, methodVersion: Int, userInfo: UserInfo)(op: (AgoraEntity) => Future[PerRequestMessage]): Future[PerRequestMessage] = {
-    // TODO add Method to model instead of exposing AgoraEntity?
-    methodRepoDAO.getMethod(methodNamespace, methodName, methodVersion, userInfo) flatMap { agoraEntityOption => agoraEntityOption match {
-      case None => Future.successful(RequestComplete(ErrorReport(StatusCodes.NotFound, s"Cannot get ${methodNamespace}/${methodName}/${methodVersion} from method repo.")))
-      case Some(agoraEntity) => op(agoraEntity)
-    }} recover { case throwable =>
-        RequestComplete(ErrorReport(StatusCodes.BadGateway, s"Unable to query the method repo.",methodRepoDAO.toErrorReport(throwable)))
-    }
+    toFutureTry(methodRepoDAO.getMethod(methodNamespace, methodName, methodVersion, userInfo)) flatMap { agoraEntityOption => agoraEntityOption match {
+      case Success(None) => Future.successful(RequestComplete(ErrorReport(StatusCodes.NotFound, s"Cannot get ${methodNamespace}/${methodName}/${methodVersion} from method repo.")))
+      case Success(Some(agoraEntity)) => op(agoraEntity)
+      case Failure(throwable) => Future.successful(RequestComplete(ErrorReport(StatusCodes.BadGateway, s"Unable to query the method repo.",methodRepoDAO.toErrorReport(throwable))))
+    }}
```

Comment:
the change above was because the recover code was erroneously handling in op(agoraEntity) and turning exceptions into bad gateway agora errors